File: crescer/java/trabalhos/tcc/api/src/main/java/br/com/cwi/crescer/api/service/security/FindUserAuthenticatedService.java
package br.com.cwi.crescer.api.service.security;

import br.com.cwi.crescer.api.model.UserAccount;
import br.com.cwi.crescer.api.repository.UserAccountRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Service;

@Service
public class FindUserAuthenticatedService {
    @Autowired
    private UserAccountRepository userAccountRepository;

    public UserAccount getUser(){
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        SecurityUser securityUser = (SecurityUser) authentication.getPrincipal();
        return  userAccountRepository.findById(securityUser.getId()).get();
    }
}

File: back/healthCare/src/main/kotlin/com/healthCare/api/user/FindUserAuthenticatedService.kt
package com.healthCare.api.user

import com.healthCare.api.model.UserAccount
import com.healthCare.api.repository.UserAccountRepository
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Service


@Service
class FindUserAuthenticatedService(
    private val userAccountRepository: UserAccountRepository
    ) {

    fun getUser(): UserAccount {
        val authentication = SecurityContextHolder.getContext().authentication
        val securityUser = authentication.principal as SecurityUser
        return userAccountRepository.findById(securityUser.id).get()
    }
}

